### DISCUSSION TITLE
Hot to get the preview of daisyUI colors in VSCode

### DISCUSSION THREAD

**Discussion by stefan-kracht** [saadeghi/daisyui#1687] [General]
I have the tailwind extension in VSCode and I like how it previews my colors. The daisy UI colors are not shown, and I kind of understand why. I just wonder if there is a way around it? I would be fine if it shows me the colors of the first theme that I have configured.

**Comment by saadeghi** [comment]
Well, the code editor (or an extension) cannot know which theme is the active theme.
By default, daisyUI provides multiple themes. `light` is the default theme for light mode and `dark` is the default theme if your OS is set to dark mode but regardless of that, you can set another theme in `<html data-theme="anything">`. So how the code editor can know which theme is the active theme to show you the color? 😁

And I think that's not a flaw. because the whole point is, you don't have to even think about the color value.   
You just use the names and the color values are handled by the current theme. For example if it's the surface, you use `base-100` colors for the background. You won't need to think if it's a light or dark. and then you use `base-content` for the text on it. again you won't need to think about the contrast. If the dark theme is active, it will be light text on dark background. If it's a light theme it will be dark text on light background. Easy!

Let me know if you have any questions.

**Comment by stefan-kracht** [comment]
Thanks for your reply! Yes, I totally get it. The intention comes more from a DX point of view. I'm used to quickly spot every color dependent classes thanks to the color preview next to it, but this does not apply here anymore. 

But I agree that this is correct, since that color values can not be statically analyzed (at least when you have more than one theme, and I have two). I was just curious if I might miss something here.
